crates/noirc_frontend/src/monomorphization/ast.rs

```diff
@@ -226,8 +226,12 @@ impl Program {
         Program { functions, main_function_signature }
     }
 
-    pub fn main(&mut self) -> &mut Function {
-        &mut self.functions[0]
+    pub fn main(&self) -> &Function {
+        &self[Self::main_id()]
+    }
+
+    pub fn main_mut(&mut self) -> &mut Function {
```

Comment:
This is fine since its a small change, though ideally we don't make changes to code outside of the ssa_refactor module in the same PR as changes inside of the ssa_refactor